internal/controller/operatorconfigmap_controller.go

```diff
@@ -158,31 +151,24 @@ func (c *OperatorConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		),
 	)
 
+	generationChangePredicate := predicate.GenerationChangedPredicate{}
 	bldr := ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.ConfigMap{}, configMapPredicates).
 		Owns(&corev1.Service{}, servicePredicate).
+		Owns(&csiopv1a1.OperatorConfig{}, builder.WithPredicates(generationChangePredicate)).
+		Owns(&csiopv1a1.Driver{}, builder.WithPredicates(generationChangePredicate)).
 		Watches(&configv1.ClusterVersion{}, enqueueConfigMapRequest, clusterVersionPredicates).
 		Watches(&extv1.CustomResourceDefinition{}, enqueueConfigMapRequest, builder.OnlyMetadata).
 		Watches(&opv1a1.Subscription{}, enqueueConfigMapRequest, subscriptionPredicates).
 		Watches(&admrv1.ValidatingWebhookConfiguration{}, enqueueConfigMapRequest, webhookPredicates).
 		Watches(&v1alpha1.StorageClient{}, enqueueConfigMapRequest, builder.WithPredicates(predicate.AnnotationChangedPredicate{}))
 
-	generationChangePredicate := predicate.GenerationChangedPredicate{}
-	if utils.DelegateCSI {
-		bldr = bldr.
-			Owns(&csiopv1a1.OperatorConfig{}, builder.WithPredicates(generationChangePredicate)).
-			Owns(&csiopv1a1.Driver{}, builder.WithPredicates(generationChangePredicate))
-	}
-
 	return bldr.Complete(c)
 }
 
 //+kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;list;watch
 //+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get;list;watch
-//+kubebuilder:rbac:groups="apps",resources=deployments,verbs=get;list;watch;create;update;patch;delete
```

Comment:
can you please also check the access for configmap we might not need  more access to it.

Reply:
above line is for config/clusterversion if you got confused. we already have all perms for configmap setup properly.

Reply:
not likely, as there are no specific RBAC changes for configmap,i used this as a placeholder for question, we this operator is not using the configmap for cephcsi like we used to create/update/delete the CM as required, if we still need all the access to the configmap we are good, if not i though it might be the good time to visit and see do we need all permission for the configmap or not.

Reply:
got it, we still need it as we have controller for reconciling configmap. if we want to reduce rbac, overall I will do that separately.

Reply:
> we might not need more access to it.
- now that I reread your comment again, my initial response was for _**we might need more access to it.**_